### ISSUE TITLE
Undefined method `Name' for `CfnDsl::AWS::Types::AWS_SSM_Association`

### ISSUE THREAD

**Issue by ouranos** [cfndsl/cfndsl#485]
Hi,

We're trying to upgrade `cfndsl` to the latest version (after being on and old version for a while) and getting some issue with `SSM_Association`

Here's my test file to reproduce:

```ruby
CloudFormation {
  Description "Test"

  SSM_Association "Test" do
    Name "AWS-RunShellScript"
  end
}
```

### Before: cfndsl 0.17.5

```
$ cfndsl --version
cfndsl 0.17.5

# Using an old specification to not make it crash
$ cfndsl -u 2.21.0
Updating specification file
Specification successfully written to ~/.cfndsl/resource_specification.json

$ jq -c '.ResourceSpecificationVersion' ~/.cfndsl/resource_specification.json
"2.21.0"

$ cfndsl test.rb | jq

{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": "Test",
  "Resources": {
    "Test": {
      "Properties": {
        "Name": "AWS-RunShellScript"
      },
      "Type": "AWS::SSM::Association"
    }
  }
}
```

Note if removing the downloaded specs and using the bundled one, it still works:

```
$ jq -c '.ResourceSpecificationVersion' "$(bundle show cfndsl)/lib/cfndsl/aws/resource_specification.json"
"1.9.1"
```

### cfndsl 1.5.0

```
$ cfndsl -u 
Updating specification file
Specification 106.0.0 successfully written to ~/.cfndsl/resource_specification.json

$ jq -c '.ResourceSpecificationVersion' ~/.cfndsl/resource_specification.json
"106.0.0"

$ cfndsl --version
cfndsl 1.5.0

$ cfndsl test.rb
undefined method `Name' for #<CfnDsl::AWS::Types::AWS_SSM_Association:0x00007fe2ee0555e0> (NoMethodError)
```

But the specification file has the `Name` as a property:
```
$ jq -c '.ResourceTypes["AWS::SSM::Association"].Properties | keys' ~/.cfndsl/resource_specification.json

# 2.21.0
["AssociationName","DocumentVersion","InstanceId","Name","OutputLocation","Parameters","ScheduleExpression","Targets"]

# 106.0.0

["ApplyOnlyAtCronInterval","AssociationName","AutomationTargetParameterName","CalendarNames","ComplianceSeverity","DocumentVersion","InstanceId","MaxConcurrency","MaxErrors","Name","OutputLocation","Parameters","ScheduleExpression","ScheduleOffset","SyncCompliance","Targets","WaitForSuccessTimeoutSeconds"]
```

Getting the same problem from `1.0.x` onward and with both versions of the specifications.

**Comment by ouranos** [comment]
Comparing the built in specs for the two versions:
```
➜ jq '.ResourceTypes["AWS::SSM::Association"].Properties.Name' ~/.gem/ruby/2.7.7/gems/cfndsl-0.17.5/lib/cfndsl/aws/resource_specification.json
{
  "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ssm-association.html#cfn-ssm-association-name",
  "PrimitiveType": "String",
  "Required": true,
  "UpdateType": "Immutable"
}
```

```
➜ jq '.ResourceTypes["AWS::SSM::Association"].Properties.Name' ~/.gem/ruby/2.7.7/gems/cfndsl-1.5.0/lib/cfndsl/aws/resource_specification.json
{
  "Documentation": "http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ssm-association.html#cfn-ssm-association-name",
  "UpdateType": "Mutable",
  "Required": true,
  "PrimitiveType": "String"
}
```

It just change from Immutable to Mutable

**Comment by gergnz** [comment]
Thanks @ouranos - I'm just back from holidays, so might take me a while to look into this.

**Comment by ouranos** [comment]
> Thanks @ouranos - I'm just back from holidays, so might take me a while to look into this.

No worries, let me know if there's anything I can do to help.

**Comment by gergnz** [comment]
@ouranos can you please delete:
`lib/cfndsl/aws/patches/500_SSM_AssociationName_patch.json`
and let me know if that works.

This patch expects that the input parameter is `DocumentName` not `Name`. 

https://github.com/cfndsl/cfndsl/issues/385

These were dragged in verbatim, so I'm not sure what the intent was for this patch. The output with `DocumentName` as a parameter includes `DocumentName` not `Name`. I can not find anything in the CloudFormation documentation that suggests it's a valid parameter.

Thus 🤷🏽 ... I think deleting this *should* be the right thing to do here.

**Comment by ouranos** [comment]
Yes, it worked 🎉 

There are two "Name" properties:
- `Name`: (required) The name of the SSM document to apply
- `AssociationName`: (Optional) The descriptive name for the association.

This can be confusing as usually `Name` is to set the name of the resource being created. So I think that's what the patch was trying to achieve (`"rename +Name+ to +DocumentName+ to clearly indicate what the attribute is for"`) but as pointed out this is not working. And I reckon we should stick with the official Cloudformation specs.

Reference: [`AWS::SSM::Association` Properties](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ssm-association.html#aws-resource-ssm-association-properties)

**Comment by ouranos** [comment]
@gergnz Any update on this? Would be good to get a new version released with the fix.

**Comment by gergnz** [comment]
Sorry @ouranos. life got in the way...

I'll push out an update today, hopefully this morning.

**Comment by gergnz** [comment]
Released in 1.6.0
